sdk/storage/storage-blob/test/perfstress/track-2/download.spec.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PerfStressTest, PerfStressOptionDictionary } from "@azure/test-utils-perfstress";
+
+import { BlobServiceClient, StorageSharedKeyCredential } from "../../../src";
+
+// Expects the .env file at the same level as the "test" folder
+import * as dotenv from "dotenv";
+dotenv.config();
+
+interface StorageBlobDownloadTestOptions {
+  size: number;
+}
+
+const account = process.env.ACCOUNT_NAME || "";
+const accountKey = process.env.ACCOUNT_KEY || "";
+
+const sharedKeyCredential = new StorageSharedKeyCredential(account, accountKey);
+
+const blobServiceClient = new BlobServiceClient(
+  // When using AnonymousCredential, following url should include a valid SAS or support public access
+  `https://${account}.blob.core.windows.net`,
+  sharedKeyCredential
+);
+const containerName = `newcontainer${new Date().getTime()}`;
+const blobName = `newblob${new Date().getTime()}`;
+const containerClient = blobServiceClient.getContainerClient(containerName);
+const blockBlobClient = blobServiceClient
+  .getContainerClient(containerName)
+  .getBlockBlobClient(blobName);
+
+export class StorageBlobDownloadTest extends PerfStressTest<StorageBlobDownloadTestOptions> {
+  public options: PerfStressOptionDictionary<StorageBlobDownloadTestOptions> = {
+    size: {
+      required: true,
+      description: "Size in bytes",
+      shortName: "sz",
+      longName: "size",
+      defaultValue: 10
+    }
+  };
+
+  public async globalSetup() {
+    const createContainerResponse = await containerClient.create();
+    console.log(
+      `Create container ${containerName} successfully`,
+      createContainerResponse.requestId
+    );
+
+    // Create a blob
+    const uploadBlobResponse = await blockBlobClient.upload(
+      Buffer.alloc(this.parsedOptions.size.value!),
+      this.parsedOptions.size.value!
+    );
+    console.log(`Uploaded block blob ${blobName} successfully`, uploadBlobResponse.requestId);
+  }
+
+  public async globalCleanup() {
+    const deleteContainerResponse = await containerClient.delete();
+    console.log(
+      `Deleted container ${containerName} successfully`,
+      deleteContainerResponse.requestId
+    );
+  }
+
+  async runAsync(): Promise<void> {
+    await blockBlobClient.download(0);
```

Comment:
What does the parameter `0` mean?  And does this code ensure all bytes in the blob are read by the client?

Reply:
`0` is the offset, we don't need to provide it in track-2 test. It is mandatory in track 1 though.
The second param `count` takes care of the amount of data to be downloaded, if undefined, it'll download till the end.

Reply:
I understand the `0` parameter now.  But does the perf test need to do something with the returned stream for the test to be valid?  For example, this sample calls `streamToBuffer()` which I presume ensures every byte is read:

https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/samples/typescript/src/basic.ts#L81-L85

Reply:
We may not need (or want) to convert the bytes to a string, but we do need to ensure the test reads every byte (even if it just reads the byte and throws it away).

Reply:
Added streamToBuffer